Add guide for organization OAuth2 application management (#5048)
wso2/docs-is: en/includes/guides/organization-management/organization-applications.md

# Manage OAuth2 applications within organization

Applications can be created directly in the organizations and these applications can be used to consume the API resources
from the organizations. 

These applications can be created under the following conditions.

- Application's protocol is OAuth2.
- Only client credentials, password and refresh grant types can be used.

Alongside with the Application Management, following capabilities are now available in the organizations

- Role management : Roles can be managed directly in the organization and can associate with the applications
which are managed directly in the organization.

> - Inherited API Resources : The Organization API Resources and the API Resources that are created in the

Reviewer: ```suggestion
- Inherited API Resources : The {{ product_name }} defined Organization API Resources and the API Resources that are created in the
```

Author: Done

root organization are now inherited to the organization. These API Resources can be authorized to the 
applications which are managed directly in the organization.

## Inherited API Resources

The Organization API Resources and the API Resources that are created in the root organization are 

> inherited to the organizations directly. These API Resources are available in read only mode from the 
> Console application.

Reviewer: ```suggestion
inherited to the organizations directly. These API Resources are available in read only mode through the {{ product_name}} Console.
```

Author: Done

![Inherited default API Resources]({{base_path}}/assets/img/guides/applications/organization-applications/inherited-api-resources.png){: style="display: block; margin: 0; border: 0.3px solid lightgrey;"}

## Role Management

Organizations can have their own roles and manage them independently. These roles can only be associated with the 
applications that are managed directly in the organization. 

![Role management in organization]({{base_path}}/assets/img/guides/applications/organization-applications/role-management-organization.png){: style="display: block; margin: 0; border: 0.3px solid lightgrey;"}

Please refer [Manage roles]({{base_path}}/guides/users/manage-roles/) for more details on how you can manage the roles.

### Role types in organization

Organization now have two types on roles.

> - Shared roles : Roles which are created in the organization when an application is shared with the particular organization.

Reviewer: ```suggestion
- Shared roles : Roles defined by the root organization and used by B2B applications shared with this organization are created as shared roles within the organization.
```

Author: Done

- Roles managed directly in the organization : Roles which can be managed directly in the organization.

Both these roles can be assigned to any user in the organization but in the authorization, shared applications will only 
consider the shared roles to provide the authorization.

There might be scenarios where the above two types of roles can lead to a conflict of role creation. Please refer 
[Manage role conflicts in organization]({{base_path}}/guides/organization-management/manage-conflicts-in-organizations/#manage-role-conflicts-in-organization) for more information.

## OAuth2 Application Management

Organizations can have their own OAuth2 applications and manage them independently. These applications can issue tokens 
and allow access to the API Resources which belongs to the particular organization. Organization application can be 
created with the below conditions.

- Application's protocol is OAuth2.
- Only client credentials, password and refresh grant types can be used.

> Organization application can be created by using the [Organization application Add]({{base_path}}/apis/organization-apis/organization-application-mgt/#tag/Applications/operation/createApplication) API request. A bearer token with the proper scopes 

Reviewer: ```suggestion
Organization application can be created by using the [Organization application Add]({{base_path}}/apis/organization-apis/organization-application-mgt/#tag/Applications/operation/createApplication) API request. A bearer token with the proper scopes 
```

mention the reqquired scope

Author: Done

which was issued for the corresponding organization will be needed to invoke this API.

> These created applications can be edited from the Console application. 

Reviewer: ```suggestion
These created applications can be edited from the {{ product_name }} Console. 
```

Author: Done

![Organization Application Edit]({{base_path}}/assets/img/guides/applications/organization-applications/organization-application-edit.png)

> 

Reviewer: Need to mention the DCR endpoint as well

> Following operations are possible for organization applications.

Reviewer: ```suggestion
The following operations are supported for organization applications.
```

Author: Done

- Protocol level configurations
- User attributes related configurations
- API Authorization for organization application
- Role management for organization application

### Token generation from organization applications

- Password Grant

!!! abstract
    **Request format**
    ```
    curl --user <OAUTH_CLIENT_KEY>:<OAUTH_CLIENT_SECRET> -k -d "grant_type=password&username=<USERNAME>&password=<PASSWORD>" -H "Content-Type: application/x-www-form-urlencoded" https://<IS_HOST>:<IS_PORT>/t/<TENANT_DOMAIN>/o/<ORG_ID>/oauth2/token
    ```
    ---
    **Sample request**
    ```curl

>     curl --user 7wYeybBGCVfLxPmS0z66WNMffyMa:WYfwHUsbsEvwtqmDLuaxF_VCQJwa -k -d "grant_type=password&username=Charlie&password=jG9A5KrX" -H "Content-Type: application/x-www-form-urlencoded" https://localhost:9443/t/carbon.super/o/7e98b86f-63c7-41a1-8c56-c909a21a2615/oauth2/token

Reviewer: ```suggestion
    curl --user 7wYeybBGCVfLxPmS0z66WNMffyMa:WYfwHUsbsEvwtqmDLuaxF_VCQJwa -k -d "grant_type=password&username=Charlie&password=jG9A5KrX" -H "Content-Type: application/x-www-form-urlencoded" https://localhost:9443/t/carbon.super/o/7e98b86f-63c7-41a1-8c56-c909a21a2615/oauth2/token
```

please template the hostname part and root org name part to be changed for IS and Asgardeo separately.
Update other places as well

Author: Done
